### DISCUSSION TITLE
Dynamically load and uninstall a library containing Avalonia components

### DISCUSSION THREAD

**Discussion by Maklith** [AvaloniaUI/Avalonia#13961] [Q&A]
I want to dynamically load and uninstall a library containing Avalonia components in an Avalonia project, but I notice that Avalonia components seem to register a lot of things in the main project, is there any way I can remove these registrations so that they can be uninstalled

If don't create Avalonia controls from the library, everything is fine

More info #13935

**Comment by timunie** [comment]
Maybe check out how orher control libs work. Check awsome avalonia for inspiration

**Reply by Maklith** [reply]
![image](https://github.com/AvaloniaUI/Avalonia/assets/42534870/44218bd9-2c53-4a4d-a859-f95028498950)
Why is it still impossible to uninstall completely?
Do you have any thoughts on this problem?
